src/components/dialogs/PinCodeDialog.tsx

```diff
@@ -0,0 +1,140 @@
+import InputMask from 'react-input-mask'
+import React, {
+  ChangeEvent,
+  KeyboardEvent,
+  useEffect,
+  useRef,
+  useState
+} from 'react'
+
+import TextField from 'cozy-ui/transpiled/react/MuiCozyTheme/TextField'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import { Button } from 'cozy-ui/transpiled/react/Button'
+import { Dialog } from 'cozy-ui/transpiled/react/CozyDialogs'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+
+interface PinCodeDialogProps {
+  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>
+  setPinCode: (pinCode: string) => void
+}
+
+const validatePinCode = (pinCode: string): boolean =>
+  pinCode.replace(/_/g, '').length === 4
+
+export const PinCodeDialog = ({
+  setPinCode,
+  setModalVisible
+}: PinCodeDialogProps): JSX.Element => {
+  const [confirmValue, setConfirmValue] = useState('')
+  const [error, setError] = useState('')
+  const [flowStep, setFlowStep] = useState<'initial' | 'confirm'>('initial')
+  const [initialValue, setInitialValue] = useState('')
+  const onCancel = (): void => setModalVisible(false)
+  const inputRef = useRef<HTMLInputElement>()
+  const { t } = useI18n()
+
+  const handleChange = (
+    event: ChangeEvent<HTMLInputElement> | KeyboardEvent
+  ): void => {
+    if ('value' in event.target) {
+      flowStep === 'initial'
+        ? setInitialValue(event.target.value)
+        : setConfirmValue(event.target.value)
+    }
+
+    if ('key' in event && flowStep === 'initial') {
```

Comment:
why do we check `'key' in event` here? We don't use it after. Is that a trick to detect if the change comes from keyboard input or something similar?

Reply:
Yes that allows us to know (and TypeScript too) that the event type is discriminated to be a keyboardinput here